module/product/src/Infrastructure/Grid/ProductGridBuilder.php

```diff
@@ -141,7 +141,7 @@ public function build(GridConfigurationInterface $configuration, Language $langu
                 'parameters' => ['language' => $language->getCode(), 'product' => '{id}'],
                 'method' => Request::METHOD_DELETE,
             ],
-        ]))->orderBy('index', 'DESC');
+        ]))->orderBy('esa_index', 'DESC');
```

Comment:
This is done via a different commit already. Perhaps merge would be in place if needed